### PR TITLE
add aize specific arguments to oidc client

### PR DESCRIPTION
**Pull request by maur1**

### CODE REVIEW

File: src/configs.py [hunk 1]
```diff
@@ -143,11 +161,12 @@ def verify_schedule_credentials(cls, values):
         if values["schedule_file"] is None:
             return values
         # A valid schedule file is given; schedule-credentials are thus required:
-        c_secret, c_id, t_id = values["client_secret"], values["client_id"], values["tenant_id"]
-        if None in [c_secret, c_id, t_id]:
+        c_secret, c_id, token_url = values["client_secret"], values["client_id"], values["token_url"]
+        if None in [c_secret, c_id, token_url]:
```
Comment (haakonvt) [inline src/configs.py:165]: Bra catch!

Comment (haakonvt) [inline src/configs.py:165]: Can `token_url` ever be `None`?

File: src/configs.py [hunk 2]
```diff
@@ -93,13 +95,26 @@ def credentials(self) -> Dict[str, str]:
     def experimental_client(self):
         return create_oidc_client_from_dct(self.dict(by_alias=False))
 
+    @root_validator(skip_on_failure=True)
+    def verify_oidc_params(cls, values):
+        if values["token_url"] is None:
+            tenant_id = values["tenant_id"]
+            values["token_url"] = f"https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token"
```
Comment (haakonvt) [inline src/configs.py:102]: Should we give a warning if both `tenant_id` and `token_url` are passed?

**Review by haakonvt — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -18,6 +18,13 @@ This action deploys a Python function to Cognite Functions, optionally with sche
 1. `schedules_client_id`:  Client ID to be used at RUNTIME for the function, but ONLY for its scheduled runs!
 1. `schedules_tenant_id`:  Tenant ID to be used at RUNTIME for the function, but ONLY for its scheduled runs!
 
+#### Required *If using a custom identity provider*
```
Comment (haakonvt) [inline README.md:21]:
```suggestion
#### May be required if using a custom identity provider
```

File: src/configs.py (hunk 1 above)
Comment (maur1) [reply src/configs.py:165]: Actually not, good catch. Will remove the check

**Review by haakonvt — APPROVED** [review]
Lgtm! 🚀  Please have a final look at my few last commits, then merge using **squash**! 🤠

**Comment by haakonvt** [comment]
No longer needed for Aize

**Pull request closed without merging**